### DISCUSSION TITLE
[1주차 그리디] 회고

### DISCUSSION THREAD

**Discussion by bigsaigon333** [fe-backpackers/js-algorithm-basecamp#15] [Announcements]
# 알고리즘 스터디 회고
- 주차: 1주차
- 주제: 그리디
- 일시: 2021-10-03 Sun 22:00 ~ 23:00

## 회고 순서
- 체크인
   -  현재 기분을 10점 만점으로 표현 및 이유 기재
- 조 배정: 3인 1조
- 리뷰: 약 30분
  - 조원의 PR에 코멘트 남기기: 모르는 내용을 물어봐요. 궁금한 점을 질문해요. 대단한 게 있으면 칭찬해요.
  - 내 PR 코멘트에 답변 달기
  - 시간이 남으면 같은 조 외의 스터디원꺼도 리뷰해요
  - 다른 스터디원이 학습한 걸 자기껄로 흡수해요
  - 코멘트 받은 내용을 정리해서 레포에 반영해요
- 회고 
   - 이번 주차 스터디하면서 어려웠던 점, 좋았던 점 서로 공유
   - 오늘 회고 때 가장 인사이트가 있었던 점 공유
   - 건의사항: 스터디 건의사항
   - action plan: 회고를 통해 얻은 다음 주차 스터디 때 적용하고 싶은 행동 방안 공유
- 체크아웃
   - 현재 기분을 10점 만점으로 표현 및 이유 기재

**Comment by bigsaigon333** [comment]
## 1. 체크인
현재 기분을 10점 만점으로 표현하고 이유를 간단히 기재해주세요 :dog2:

**Reply by bigsaigon333** [reply]
동동: 8점 첫 회고시간을 주최하게 되어서 매우 떨리네요! 많은 분들이 들어와주셔서 긴장됩니다 :sweat_drops:

**Reply by skt-t1-byungi** [reply]
리밋: 낼쉰다고 생각하면 7 낼모래 생각하면 6; 저는 와리가리합니다 ㅎ

**Reply by moonheekim0118** [reply]
문희: 8점 운동하고 와서 개운합니다 2점은 한문제 못풀어서 찝찝한 마음에 깎았어요 ㅎㅎ

**Reply by Sunmon** [reply]
선정: 7 대창이 맛있었습니다

**Reply by Yeonlisa** [reply]
연리사 : 7점 챕터11을 손도못댔네요

**Reply by ganeodolu** [reply]
호준: 7점, 제대로 다 훑어보지 못한 점과 새로 스터디 시작하는 마음이 합쳐서 7점입니다 :smile:

**Reply by sooster910** [reply]
Sue: 8점, 해답을 봐야 이해를 하는게  아쉽습니다.

**Reply by saltwoodYJ** [reply]
YJ :8점, 공부를 제대로 끝마치지못해서 죄송한마음과 배워간다는 감사한 마음에 8점입니다!ㅎㅎ

**Reply by hyunhaha** [reply]
HH: 7점 좋은 스터디를 찾아서 좋지만 늦게 들어오게 되서 조금 아쉬운 마음입니다! 그리고 pr남기기는 처음이라 헤메는 시간도 있었어요 ㅎ;;

**Reply by ddongule** [reply]
그루밍: 8점 ㅠㅠ 오늘 많이 쉬어서 넘 좋았는데 문제를 많이 못풀어서 아쉬우어요오 그래서 2점 깎았숩니다

**Comment by bigsaigon333** [comment]
## 회고
- 이번 주차 스터디하면서 어려웠던 점, 좋았던 점 서로 공유
- 오늘 회고 때 가장 인사이트가 있었던 점 공유
- 건의사항: 스터디 건의사항
- action plan: 회고를 통해 얻은 다음 주차 스터디 때 적용하고 싶은 행동 방안 공유

**Reply by moonheekim0118** [reply]
### 좋았던 점 
- 혼자 여러 문제 풀던 것 보다, 그리디라는 주제에 집중해서 문제를 풀 수 있었어서 너무 좋았어요! 
- 알고리즘 문제 푸는게 동기부여가 잘 안되었는데 스터디 참여자체가 동기부여가 되서 좋았습니다 ㅎㅎ

### 어려웠던 점
- 아직 답지를 안보면 못푸는 문제들이 있었어서 ㅠ.ㅠ..그점이 어려웠습니다.

### 오늘 회고 때 가장 인사이트가 있었던 점
- 스터디원분들 코드에서 함수 재사용이나 util 함수 작성을 목격(?) 한 점 !! 저는 알고리즘 문제에는 깔끔한 코드를 적용한다기 보다 일단 문제를 푼다!!라는 생각으로 풀어왔는데..깔끔한 코드를 보고 많이 반성했습니다ㅎㅎ
- reduce, map, forEach와 같은 자바스크립트 메서드들을 잘 사용해주신 점이 인상깊었습니다

### 액션 플랜 
- 자바스크립트의 메서드를 잘 사용하여 문제를 푼다!!
- util 함수를 작성해서 함수를 최대한 재사용한다 !!
- 시간 복잡도를 계산한다 !!

**Reply by bigsaigon333** [reply]
**동동**

- 처음 하는 회고 시간이라 많이 긴장이 되었습니다. 구글 밋, 줌, 디스코드 사용하면서 음향에 문제가 있었던 적은 없었는데, 오늘 노이즈가 너무 심하다고 하셔서 그때부터 멘탈이 붕괴되어 버렸네요 ㅎㅎ
- 남의 코드 읽는게 확실히 시간이 걸리네요. 특히 알고리즘 같은 경우에는, 가독성보다는 문제 풀이에 집중을 하면서 코드를 작성하다보니, 다른 코드보다 좀 더 읽기 어렵게 느껴지는 거 같아요 

## 좋았던 점
- solution 함수와 같이 깔끔하게 작성해주니 확실히 보기 편한 것 같아요. 그리고 jsdoc comment도 깔끔하게 작성해주셔서 더욱 그렇구요


## 액션 플랜
- 다른 스터디원이 내 코드가 제대로 동작하는지 손쉽게 돌려볼 수 있게끔 작성하자
- 음성이 지지직 거리는 현상을 해결하자

## 건의사항
- 리뷰 시간이 확실히 부족한 느낌인데, 
   1. 사전에 조를 짜서 리뷰를 해오는 방식이 좋을까요?
   2. 아니면 리뷰시간을 좀 길게 가질까요? 
 의견 수렴해서 회고 방식을 좀 더 다듬어보고 싶네요

**Reply by ganeodolu** [reply]
좋았던 점 : `~~` 연산자를 내림으로써 활용하는 것에 대해 알게 됨, 스터디로 다양한 피드백을 받을 수 있는 점
어려웠던 점 : 풀었던 것도 다시 보니 머리가 리셋되었던 점
건의사항 : 아직 모르겠음
action plan : PR 리뷰 받은 것 보고, 반영해보기

**Reply by Sunmon** [reply]
- 좋았던점: 다른 분들의 코드를 보면서 놓쳤던 부분을 볼 수 있어서 좋았습니다. 그냥 버릇처럼 썼던 함순데 간단하게 표현할 방법이 있었네요! docs도 다른분꺼 보면서 좋은 부분은 차용하려구요 ㅎㅎ
- 어려웠던점: 코드를 급하게 읽으려니 이해하기가 어렵네요 😢 좀 더 익숙해지면 나아지겠죠
- 건의사항: 다른 분이 말씀하셨던것처럼 조를 미리 정하는것도 나쁘지 않을 것 같습니다
- action plan: 혹시 풀다가 안 풀리는 문제가 있다면 여쭤보고 싶은데 디코에 공유해도 될까요?
  - 테스트케이스 만들면 공유하기
  - 문제 공유하기
  - 주석으로 수도코드 작성하기

**Reply by skt-t1-byungi** [reply]
모두 프로그래머스 포맷을 사용해주셔서 작성하면서 맹든 테스트케이스 돌려보기 좋았습니다
책내용과 다르게 풀었을때 (특히 긴코드일때) 작성자의 해설없이 코드를 짧은시간내 이해하기 쉽지 않네요
혼자공부하면 텐션을 유지하기 힘든데요, 누군가 지켜볼지 모른다는 긴장감만으로도 스터디 충분히 의의가 있찌 않나 생각이 들었슴다 ㅎ
### 그리고 책의 테스트케이스가 너무 적어서 테스트케이스를 서로 공유했으면 좋겠네요 ㅎㅎ

**Reply by ddongule** [reply]
### 좋았던 점
- 이렇게 알고리즘을 다른 분들과 함께 푼다고 생각하니 든든하네요! 의지도 조금 더 생기게 되구요
### 어려웠던 점
- 이번 주가 좀 정신이 없어서 문제를 많이 못풀었다는 점이 아쉽습니다.. 다음 주는 더 열심히 할게요.. ㅠㅠ
### 인사이트
- 아직 인사이트가 없네요.. 제가 공부를 안해와서.. 나자신.. 반성하자..
### 건의사항
- 지금 이 상태두 저는 좋아요 😎
### Action Plan
- 다음 주부터는 꾸준히, 매일 커밋하는 것을 목표로 해보겠습니다..!

**Reply by Yeonlisa** [reply]
좋았던 점 : 몰랐던 연산자라던지, 모듈같은건 물론이고 좀더 직관적으로 이해할 수 있는 로직을 보면서 놀라운 시간이었습니다
어려웠던 점 : 코드해석이 힘들어졌다
액션 플랜 : 수도코드를 주석으로 달아보자

**Reply by bigsaigon333** [reply]
@Sunmon 
> * action plan: 혹시 풀다가 안 풀리는 문제가 있다면 여쭤보고 싶은데 디코에 공유해도 될까요?

그럼요~~!!!! 그러려고 만든 디스코드입니다 ㅎㅎ

**Reply by saltwoodYJ** [reply]
- 좋았던점 : 아직 pr하는 방법이 미숙해서 스터디 준비를 못해왔으나 감사하게도 참관할수있었습니다. 다른분들이 정리하신 내용과 코드들을 통해 이해가 어려웠던 부분들에 대해서 도움을 받을수 있어서 좋았습니다 ㅎㅎ
- 어려웠던 점 : 스터디주제에 대해서 이해도가 낮은채로 참여하게 되어서 제대로된 리뷰가 불가능했습니다. 다음 스터디부터는 준비를 제대로 해오도록 하겠습니다
- action plan : pr보내는 방법과 review 하는 방법에 대해 다시 연구해 다음부터는 제대로된 pr review를 할수있도록 해야겠습니다ㅎㅎ

**Reply by sooster910** [reply]
- 역시 다른 분들의 코드를 보면서 생각치 못한 접근, 문법을 사용하는 것을 보면서 감탄을 하게 됩니다. 하길 잘했다는 생각이 듭니다. 
- 다른분들의 코드를 보면서 같이 이야기를 나누고 싶었는데 , 코드를 읽는데 시간이 걸리다보니 아쉽습니다. 그리고 풀었던 문제도 다른 분의 코드를 보니.. 다시어렵습니다. 그만큼 다양한 코드에 익숙해져야 할 필요성이 느껴집니다.
- 다음 주에는 중간중간 여유있게 다른 분들의 코드를 리뷰하면서 여쭤봐야 해야겠습니다. 그만큼 저도 미리 공부를 해야할 것같습니다.

**Reply by esther615kim** [reply]
judy.k
- 알고리즘 스터디는 처음이라 깃허브 PR부터 헤매고 있었어요 
- 그래도 스터디원분들의 좋은 코드들을 많이 봐서 눈이 맑아짐

액션플랜
- 이번주는 PR 하기가 겁나서.. 혼자서 교재랑 문제풀이(코드샌드박스로)해봤는데 
  다음주부터는 PR 참여까지를 목표로 해보겠습니다!

질문
- 알고리즘 문제풀이에서 네이밍은 짧고 간단히 작성하는 편인지 궁금했습니다.
 기존에는 최대한 정확하게 네이밍하라는 코드리뷰를 받아서 길게길게 쓰고 있었어요 => 
 예를 들어서 이번 그리디 문제에서는 makeMinimumChange,makeBiggestNumber, makeNumber 이런식으로...

**Reply by hyunhaha** [reply]
-  깃허브를 이용해서 스터디하는 것은 처음인데 알고리즘 공부뿐만 아니라 깃허브 사용도 공부할 수 있는 기회여서 좋습니다!
- js로 된 코딩테스트 스터디를 찾기 어려운데 너무 적절한 주제의 스터디를 찾아서 좋습니다
- 다른 분들의 다양한 코드쓰는 방법을 보고 배울 수 있고 동기부여가 되어서 좋습니다

**Comment by bigsaigon333** [comment]
## 체크아웃
현재 기분을 10점 만점으로 표현 및 이유 기재

**Reply by moonheekim0118** [reply]
10점 !! 여러 코드보면서 공부하고, 많은 분들과 소통해서 즐거웠습니다

**Reply by ganeodolu** [reply]
8점 : 스터디하면서 알고리즘에 관한 얘기 나눌 수 있어서 좋았습니당

**Reply by ddongule** [reply]
10점! 같이 이렇게 이야기하니까 재밌고 좋네요 ㅎㅎ 의지도 다질 수 있어서 넘 좋았어요!!!!

**Reply by Sunmon** [reply]
9점: 다양한 코드를 보고 소통할 수 있어 좋았습니다 👍

**Reply by bigsaigon333** [reply]
**동동** 

9점: 역시 사람들과 소통하는 건 많은 인사이트를 얻게 되네요! 좋은 에너지도 얻어 갑니다!
앞으로 어떻게 더 스터디를 발전시켜 나갈지에 대한 몇가지 사안을 알게 되어서, 이부분에 대해서 또 고민해보고 의견 수렴하고 적용하고 해 나갈 생각입니다! 모두 화이팅~!

**Reply by Yeonlisa** [reply]
8점 : 코드보면서 배우고 많은분들과 소통할 수 있어서 좋았습니다

**Reply by skt-t1-byungi** [reply]
8점: 좀있다 잘거라서 ㅎㅎ

**Reply by saltwoodYJ** [reply]
10점 : 다음스터디는 오늘보다 더 잘할수있을거같다는 느낌에 10점입니다! ㅎㅎ

**Reply by hyunhaha** [reply]
9점: 다양한 코드를 보고 소통하며 동기부여 되는 기회여서 좋았습니다!
